tests/conftest.py

```diff
@@ -69,16 +73,25 @@ def os_agnostic_snapshot(snapshot):
 #
 # This automatically used setup fixture is required to use test.conf from resources
 # in unit tests which are not using "runner" fixture (tests which do not invoke CLI command).
+#
+# In addition to its own CliContextManager, each test gets its own OpenConnectionCache
+# which is cleared after the test completes.
 def reset_global_context_and_setup_config_and_logging_levels(
     request, test_snowcli_config
 ):
-    cli_context_manager = get_cli_context_manager()
-    cli_context_manager.reset()
-    cli_context_manager.set_verbose(False)
-    cli_context_manager.set_enable_tracebacks(False)
-    config_init(test_snowcli_config)
-    loggers.create_loggers(verbose=False, debug=False)
-    yield
+    with fork_cli_context():
+        connection_cache = OpenConnectionCache()
+        cli_context_manager = get_cli_context_manager()
+        cli_context_manager.reset()
+        cli_context_manager.verbose = False
+        cli_context_manager.enable_tracebacks = False
+        cli_context_manager.connection_cache = connection_cache
+        config_init(test_snowcli_config)
+        loggers.create_loggers(verbose=False, debug=False)
```

Comment:
there are a few things set here that aren't being set in the fork for integration tests, is that intentional?

Reply:
This preserves the existing behaviour. @sfc-gh-turbaszek any thoughts on differences between context manager setup for unit vs integration tests?